### DISCUSSION TITLE
Multiple clients with single Kratos UI

### DISCUSSION THREAD

**Discussion by sergeyshaykhullin** [ory/kratos#996] [Q&A] [answered]
Can you please help me to find docs about multiple client working with Kratos using same ui?

Kratos+node example has:

```
Client UI + Registration/Login UI is same app  <----->  Kratos API
```

Does Kratos supports something like OIDC SSO flows?

```
Client1 UI <---->
                   Kratos API  <--> Registration/Login UI
Client2 UI <---->
```

**Comment by zepatrik** [comment, marked as answer]
If your "clients" are all first party applications that you trust, you can add them to the allow list and set the `return_to` query parameter when initializing a flow. Some related topics you should look into: #662 #994 
In any other case you will need e.g. [Ory Hydra](https://github.com/ory/hydra) or something comparable.
